dagmc/dagnav.py

```diff
@@ -61,48 +62,118 @@ def category_tag(self):
         "Curve" and "Vertex" are also present in the model options but those classes are not supported in this package.
         """
-        return self.mb.tag_get_handle(types.CATEGORY_TAG_NAME)
+        return self.mb.tag_get_handle(
+            types.CATEGORY_TAG_NAME,
+            types.CATEGORY_TAG_SIZE,
+            types.MB_TYPE_OPAQUE,
+            types.MB_TAG_SPARSE,
+            create_if_missing=True
+        )
 
     @cached_property
     def name_tag(self):
-        return self.mb.tag_get_handle(types.NAME_TAG_NAME)
+        return self.mb.tag_get_handle(
+            types.NAME_TAG_NAME,
+            types.NAME_TAG_SIZE,
+            types.MB_TYPE_OPAQUE,
+            types.MB_TAG_SPARSE,
+            create_if_missing=True,
+    )
 
     @cached_property
     def geom_dimension_tag(self):
-        return self.mb.tag_get_handle(types.GEOM_DIMENSION_TAG_NAME)
+        return self.mb.tag_get_handle(
+            types.GEOM_DIMENSION_TAG_NAME,
+            1,
+            types.MB_TYPE_INTEGER,
+            types.MB_TAG_SPARSE,
+            create_if_missing=True,
+        )
 
 
 class DAGSet:
     """
     Generic functionality for a DAGMC EntitySet.
     """
-    def __init__(self, model: DAGModel, handle):
+    def __init__(self, model: DAGModel, handle: np.uint64):
         self.model = model
         self.handle = handle
 
+    def _check_category_and_dimension(self):
+        """Check for consistency of category and geom_dimension tags"""
+        stype = self._category.lower()
+        geom_dimension = self.geom_dimension
+        category = self.category
+
+        if geom_dimension != -1:
+            # If geom_dimension is assigned but not consistent, raise exception
+            if geom_dimension != self._geom_dimension:
+                raise ValueError(f"DAGMC {stype} ID={self.id} has geom_dimension={geom_dimension}.")
+
+            # If category is unassigned, assign it based on geom_dimension
+            if category is None:
+                if stype == 'group':
+                    warn(f"Assigned category {self._category} to {stype} '{self.name}'.")
+                else:
+                    warn(f"Assigned category {self._category} to {stype} ID={self.id}.")
+                self.category = self._category
+
+        if category is not None:
+            # If category is assigned but not consistent, raise exception
+            if category != self._category:
+                raise ValueError(f"DAGMC {stype} ID={self.id} has category={category}.")
+
+            # If geom_dimension is unassigned, assign it based on category
+            if geom_dimension == -1:
+                if stype == 'group':
+                    warn(f"Assigned geom_dimension={self._geom_dimension} to {stype} '{self.name}'.")
+                else:
+                    warn(f"Assigned geom_dimension={self._geom_dimension} to {stype} ID={self.id}.")
+                self.geom_dimension = self._geom_dimension
```

Comment:
Maybe just one more check here that one of `geom_dimension` or `category` are set correctly. Otherwise it's likely that the handle of the object isn't valid for use in the class.

Reply:
Good suggestion -- just added that in!